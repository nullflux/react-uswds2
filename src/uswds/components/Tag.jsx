import React from 'react';
import PropTypes from 'prop-types';


//  A tag draws attention to new or categorized content elements.
// 
//  Required props:
//  - children: node, usually text to display within the tag
// 
//  Optional props:
//  - classes: string, additional classes
 

export default class Tag extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    classes: PropTypes.string
  };

  static defaultProps = {
    classes: ''
  };

  render() {
    return (
      <span className={`usa-tag ${this.props.classes}`}>
        {this.props.children}
      </span>
    );
  }
}