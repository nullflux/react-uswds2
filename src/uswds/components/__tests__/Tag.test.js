import React from 'react';
import { shallow } from 'enzyme';
import Tag from '../Tag.jsx';

jest.unmock('../Tag.jsx');

describe('Tag', () => {
  it('should render correctly', () => {
    const component = shallow(<Tag>TestText</Tag>);
    expect(component).toMatchSnapshot();
  });
});