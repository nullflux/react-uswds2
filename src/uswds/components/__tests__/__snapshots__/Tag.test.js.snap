// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render correctly 1`] = `
<span
  className="usa-tag "
>
  TestText
</span>
`;
